### PR TITLE
tests: fix conflict between Cygwin/msys and Windows PIDs

### PR DESCRIPTION
**Pull request by mback2k**
Add 65536 to Windows PIDs to allow Windows specific treatment in POSIX Perl based test suite.

See also:
- https://cygwin.com/git/?p=newlib-cygwin.git;a=commit;h=b5e1003722cb14235c4f166be72c09acdffc62ea
- https://cygwin.com/git/?p=newlib-cygwin.git;a=commit;h=448cf5aa4b429d5a9cebf92a0da4ab4b5b6d23fe

Replaces #5178

This will hopefully fix the random issues encountered with Windows CI builds as described in #5034.

### CODE REVIEW

**Comment by mback2k** [comment]
**TODO:** I guess `secureserver.pl` (stunnel) and `sshserver.pl` (openssh) will also need to be updated.

**Comment by jay** [comment]
Just FYI Windows PIDs are often over 65535 on Enterprise systems that churn a lot of processes.
Capture

**Comment by bagder** [comment]
According to [some sources](https://stackoverflow.com/questions/17868218/what-is-the-maximum-process-id-on-windows) a Windows pid can be anything within 32bit...

**Comment by mback2k** [comment]
Yes, Windows PIDs can be anything, but Cygwin/msys PIDs are generated below 65536 at the moment. So offsetting Windows PIDs by 65536 will work until they are generated below MAX_INT.

I was initially thinking about using negative values for native Windows PIDs, but that would require changes to some 500 LoCs in the Perl scripts.

File: tests/server/util.c [hunk 1]
```diff
@@ -269,6 +269,15 @@ int write_pidfile(const char *filename)
...
     return 0; /* fail */
   }
+#if defined(WIN32) || defined(_WIN32)
+  /* store pid + 65536 to avoid conflict with Cygwin/msys PIDs, see also:
+   * - https://cygwin.com/git/?p=newlib-cygwin.git;a=commit; ↵
+   *   h=b5e1003722cb14235c4f166be72c09acdffc62ea
+   * - https://cygwin.com/git/?p=newlib-cygwin.git;a=commit; ↵
+   *   h=448cf5aa4b429d5a9cebf92a0da4ab4b5b6d23fe
+   */
+  pid += 65536;
```
Comment (MarcelRaad) [inline tests/server/util.c:279]: Can't this result in an overflow? I'm not a Perl developer, but I assume there's no overflow in Perl? Maybe use `curl_off_t`?

Comment (mback2k) [reply tests/server/util.c:279]: Thanks, I added another commit to handle this situation as you suggested.

**Pull request closed without merging**